pkg/vminfo/vminfo.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+// Package vminfo extracts information about the target VM.
+// The package itself runs on the host, which may be a different OS/arch.
+// User of the package first requests set of files that needs to be fetched from the VM
+// (Checker.RequiredFiles), then fetches these files, and calls Checker.MachineInfo
+// to parse the files and extract information about the VM.
+// The information includes information about kernel modules and OS-specific info
+// (for Linux that includes things like parsed /proc/cpuinfo).
+package vminfo
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/host"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+type Checker struct {
+	checker
+}
+
+func New(cfg *mgrconfig.Config) *Checker {
+	switch {
+	case cfg.TargetOS == targets.Linux:
+		return &Checker{new(linux)}
+	default:
+		return &Checker{new(stub)}
+	}
+}
+
+func (checker *Checker) MachineInfo(fileInfos []host.FileInfo) ([]host.KernelModule, []byte, error) {
+	files := createFilesystem(fileInfos)
```

Comment:
The proposal is to call it createVirtualFilesystem().
I expected to see the real files created here.

Reply:
Done.